### ISSUE TITLE
Huggingface-hub 0.26.0 causes ImportError 'url_to_filename' from 'huggingface_hub.file_download' when import 'LoRAConfig'

### ISSUE THREAD

**Issue by sybrenjansen** [adapter-hub/adapters#748]
## Environment info
     
- `adapters` version: 1.0.0
- `transformers` version: 4.43.4
- Platform: macOS-14.6.1-arm64-arm-64bit
- Python version: 3.10.13
- Huggingface_hub version: 0.26.0
- Safetensors version: 0.4.5
- Accelerate version: 0.34.2
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.4.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed

## Information

When importing `from adapters import LoRAConfig` I get the following error:

```python
RuntimeError: Failed to import adapters.configuration because of the following error (look up to see its traceback):
cannot import name 'url_to_filename' from 'huggingface_hub.file_download' (./venv/lib/python3.10/site-packages/huggingface_hub/file_download.py)
```

From looking at the release notes of huggingface-hub, the `url_to_filename` function has been removed and you should use a different function.

I expect that this will be an issue on different locations in the repo as well.

## To reproduce

1. `pip install -U adapters transformers huggingface-hub`
2. `python -c "from adapters import LoRAConfig"`

Giv
```python
Traceback (most recent call last):
  File "./venv/lib/python3.10/site-packages/transformers/utils/import_utils.py", line 1586, in _get_module
    return importlib.import_module("." + module_name, self.__name__)
  File "/Users/x/miniconda3/envs/py310/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "./venv/lib/python3.10/site-packages/adapters/configuration/__init__.py", line 2, in <module>
    from .adapter_config import *
  File "./venv/lib/python3.10/site-packages/adapters/configuration/adapter_config.py", line 6, in <module>
    from ..utils import resolve_adapter_config
  File "./venv/lib/python3.10/site-packages/adapters/utils.py", line 28, in <module>
    from huggingface_hub.file_download import http_get, url_to_filename
ImportError: cannot import name 'url_to_filename' from 'huggingface_hub.file_download' (./venv/lib/python3.10/site-packages/huggingface_hub/file_download.py)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1075, in _handle_fromlist
  File "./venv/lib/python3.10/site-packages/transformers/utils/import_utils.py", line 1576, in __getattr__
    module = self._get_module(self._class_to_module[name])
  File "./venv/lib/python3.10/site-packages/transformers/utils/import_utils.py", line 1588, in _get_module
    raise RuntimeError(
RuntimeError: Failed to import adapters.configuration because of the following error (look up to see its traceback):
cannot import name 'url_to_filename' from 'huggingface_hub.file_download' (./venv/lib/python3.10/site-packages/huggingface_hub/file_download.py)
```

## Expected behavior

The import statement shouldn't throw an error, but import as normal.

**Comment by 14H034160212** [comment]
I met this issue as well. After huggingface-hub 0.26.0, the `url_to_filename` has been removed. The relative code in the utils.py should be updated.
https://github.com/huggingface/huggingface_hub/releases

**Comment by lenglaender** [comment]
Thanks for reporting this critical bug. We will look into this ASAP.

**Quick fix:** Don't use huggingface_hub version 0.26 or higher. Install the latest compatible release via:  `pip install huggingface-hub==0.25`

**Comment by lenglaender** [comment]
Should be fixed once #750 is merged.
